test/Collector.test.ts

```diff
@@ -0,0 +1,101 @@
+import BurnableTestToken from '../build/BurnableTestToken.json';
+import { deployContract, MockProvider, solidity } from 'ethereum-waffle';
+
+use(solidity);
+
+describe('Collector', function () {
+    async function deployCollector(
+        share1?: number,
+        share2?: number,
+        share3?: number,
+        share4?: number
+    ) {
+        const [owner, partner1, partner2, partner3, partner4] =
+            new MockProvider().getWallets();
+        const partners = [
+            {
+                beneficiary: await partner1.getAddress(),
+                share: share1 ?? 20
+            },
+            {
+                beneficiary: await partner2.getAddress(),
+                share: share2 ?? 35
+            },
+            {
+                beneficiary: await partner3.getAddress(),
+                share: share3 ?? 13
+            },
+            {
+                beneficiary: await partner4.getAddress(),
+                share: share4 ?? 32
+            }
+        ];
+
+        const testToken = await deployContract(owner, BurnableTestToken);
+        const collector = await deployContract(owner, Collector, [
+            owner.address,
+            testToken.address,
+            partners
+        ]);
+
+        return { collector, testToken, partners, owner };
+    }
+
+    describe('Deployment', function () {
+        it('Should deploy with owner, token and revenue partners', async function () {
+            const { collector } = await deployCollector();
+            expect(collector.address).to.be.properAddress;
+        });
+
+        it('Should not let deploy with invalid shares', async function () {
+            await expect(deployCollector(25, 25, 25, 26)).to.be.revertedWith(
+                'total shares must add up to 100%'
+            );
+        });
+    });
+
+    describe('Withdraw', function () {
+        it('Should Withdraw', async function () {
+            const { collector, testToken, partners } = await deployCollector();
+            await testToken.mint(100, collector.address);
+
+            await collector.withdraw();
+            expect(partners[0].share).to.equal(
+                await testToken.balanceOf(partners[0].beneficiary)
+            );
+        });
+
+        it('Should fail when no revenue to share', async function () {
+            const { collector } = await deployCollector();
+            await expect(collector.withdraw()).to.be.revertedWith(
+                'no revenue to share'
+            );
+        });
+    });
+
+    describe('updateShares', function () {
+        it('Should update shares and partners when token balance is zero', async function () {
+            const { collector, partners } = await deployCollector();
+            await collector.updateShares(partners);
+            expect('updateShares').to.be.calledOnContract(collector);
+        });
+
+        it('Should update shares and partners after withdrawing irregular amount', async function () {
```

Comment:
what do you mean by "irregular amount"? Maybe you mean with an amount that generates reminder right?

Reply:
Yes that's what I meant

Reply:
changed to "remainder amount"